rclcpp/src/rclcpp/executors/static_single_threaded_executor.cpp

```diff
@@ -48,6 +48,8 @@ StaticSingleThreadedExecutor::spin()
     entities_collector_->refresh_wait_set();
     execute_ready_executables();
   }
+
+  entities_collector_->fini();
```

Comment:
Should fini also be called if an exception is thrown inside the while loop? That is, should this call be put in a `SCOPE_EXIT` block just after `init()`, or is it suitable just at the end here?

Reply:
Yeah, that is a good idea I didn't think of earlier.  It's pretty unlikely to happen (and if it did the test would fail, so a memory leak isn't the worst thing), but we may as well be complete while we are looking at it.  @iuhilnehc-ynos do you mind making that last change?

Reply:
@brawner 

Thank you for pointing this.
Hmmm, I think I need to catch the exception to call fini() and then rethrow it,
and also update for using SCOPE_EXIT to call fini() in the test cases.